core/object/object.h

```diff
@@ -235,7 +235,7 @@ struct MethodInfo {
 		return arguments_metadata.size() > p_arg ? arguments_metadata[p_arg] : 0;
 	}
 
-	inline bool operator==(const MethodInfo &p_method) const { return id == p_method.id; }
+	inline bool operator==(const MethodInfo &p_method) const { return id == p_method.id && name == p_method.name; }
```

Comment:
I get why you changed this given the discussion, but you're not actually using this operator in the PR, so I think it's best to not include this change.

Reply:
I understand. I can factor this change out into a separated PR.

Reply:
I removed this change and will submit it as a separate PR later.